aten/src/ATen/native/mps/operations/Sort.mm

```diff
@@ -29,10 +29,15 @@
   bool macOS13_3_plus = is_macos_13_or_newer(MacOSVersion::MACOS_VER_13_3_PLUS);
   MPS_CHECK_INT64_OP_SUPPORTED(self, macOS13_3_plus, "sort_stable_out");
 
-  values.copy_(self);
-  // check if self is scalar
-  dim = maybe_wrap_dim(dim, self.dim(), true);
-  if (self.dim() == 0 && self.numel() == 1) {
+
+  // issue #101878: MPS might give a wrong sorted tensor when dealing with a strided view tensor.
+  // create a proxy tensor to temporarily address the issue
+  Tensor input_contiguous = self.contiguous();
```

Comment:
Making it contiguous will have perf implications. can you try similar to :
https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/mps/operations/BinaryOps.mm#L84

see if this still works

Reply:
appreciate it! looking into it for a more thorough solution

Reply:
Thanks